pkg/operator/controller/machineconfig/mcp.go

```diff
@@ -106,91 +122,204 @@ func (r *MachineConfigReconciler) GetProfilingMCP() *mcv1.MachineConfigPool {
 						Kind:       MCKind,
 						Name:       CrioProfilingConfigName,
 					},
-					{
-						APIVersion: MCAPIVersion,
-						Kind:       MCKind,
-						Name:       KubeletGeneratedConfigName,
-					},
 				},
 			},
 		},
 	}
 }
 
-// createProfilingMCP is for creating the required profiling MCP
-func (r *MachineConfigReconciler) createProfilingMCP(ctx context.Context) error {
-	mcp := r.GetProfilingMCP()
+// createMCP is for creating the required MCP
+func (r *MachineConfigReconciler) createMCP(ctx context.Context, name string) error {
+	mcp := r.GetProfilingMCP(name)
 
 	if err := r.Create(ctx, mcp); err != nil {
-		return fmt.Errorf("failed to create MCP for profiling machine configs: %w", err)
+		return fmt.Errorf("failed to create MCP %s: %w", name, err)
 	}
 
 	if err := ctrl.SetControllerReference(r.CtrlConfig, mcp, r.Scheme); err != nil {
```

Comment:
This looks like a noop to me.

The resource is modified in local memory, but it's not persisted. Move it just a few lines above.

Reply:
comment incorporated